src/lib/fragments/AgGrid.react.js

```diff
@@ -949,9 +1025,29 @@ export default class DashAgGrid extends Component {
             this.rowTransaction(rowTransaction);
         }
 
+        let alignedGrids;
+        if (dashGridOptions) {
+            if ('alignedGrids' in dashGridOptions) {
+                alignedGrids = [];
+                const addGrid = (id) => {
+                    if (!agGridRefs[id]) {
+                        agGridRefs[id] = {api: null};
+                    }
+                    alignedGrids.push(agGridRefs[id]);
+                };
```

Comment:
There's a complication here with pattern-matching IDs - `props.id` will always be a string, because we handle that separately and stringify those objects when passing them from dash renderer to the actual components. But the renderer doesn't know to do that inside `alignedGrids` so we'll need to do it ourselves. And it's not just `JSON.stringify` either... we should probably export this from the renderer so it can be reused, but we don't so we'll need to copy it here. [Here's the source.](https://github.com/plotly/dash/blob/5fd9a14a05c46f5c15802109c263423930297cc8/dash/dash-renderer/src/actions/dependencies.js#L114-L123)

Reply:
Also shouldn't we do this conversion inside `convertOne`? As is it'll also be inside of `convertedProps` so it's in principle ambiguous, right?

Reply:
We cant memoize, this actually caused issues because even though the background object changed, the end result wasnt correct.

I removed it from `convertedProps`. 😄